src/components/form/form.js

```diff
@@ -514,7 +586,11 @@ class Form extends React.Component {
 
     const valid = this.validate();
 
-    if (!valid) { ev.preventDefault(); }
+    if (valid) {
+      this.resetIsDirty();
```

Comment:
not sure about this - what if it is sent through a POST and the server responds with an error, meaning the data has not been submitted? Not saying this is wrong, just wondering if you have any thoughts on it.